client.go

```diff
@@ -299,7 +301,9 @@ func (co *Conn) WriteMsg(m *Msg) (err error) {
 // Write implements the net.Conn Write method.
 func (co *Conn) Write(p []byte) (n int, err error) {
 	switch co.Conn.(type) {
-	case *net.TCPConn, *tls.Conn:
+	case *net.UDPConn:
+		break
+	case net.Conn:
 		if len(p) > MaxMsgSize {
 			return 0, &Error{err: "message too large"}
```

Comment:
actually this length check makes sense regardless of transport used.

Reply:
Agreed moved upwards.